moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp

```diff
@@ -81,19 +59,19 @@ bool HybridPlanningManager::initialize()
   }
   // TODO(sjahr) Refactor parameter declaration and use repository wide solution
   std::string logic_plugin_name = "";
-  if (this->has_parameter("planner_logic_plugin_name"))
+  if (node_->has_parameter("planner_logic_plugin_name"))
   {
-    this->get_parameter<std::string>("planner_logic_plugin_name", logic_plugin_name);
+    node_->get_parameter<std::string>("planner_logic_plugin_name", logic_plugin_name);
   }
   else
   {
-    logic_plugin_name = this->declare_parameter<std::string>("planner_logic_plugin_name",
-                                                             "moveit::hybrid_planning/ReplanInvalidatedTrajectory");
+    logic_plugin_name = node_->declare_parameter<std::string>("planner_logic_plugin_name",
+                                                              "moveit::hybrid_planning/ReplanInvalidatedTrajectory");
   }
   try
   {
     planner_logic_instance_ = planner_logic_plugin_loader_->createUniqueInstance(logic_plugin_name);
-    if (!planner_logic_instance_->initialize(HybridPlanningManager::shared_from_this()))
+    if (!planner_logic_instance_->initialize(hybrid_planner_interface_))
```

Comment:
Is `hybrid_planner_interface_` initialized at this point?  It looks to me that it isn't initialized until line 87.

Reply:
Yes :/ An uninitialized pointer is passed to the planner logic instance with the promise, that it won't be needed before it is initialized. This problem with the circular lifetime ownership I have not solved yet but I think it is better than passing the whole hybrid planning manager